.github/workflows/codeql-analysis.yml

```diff
@@ -24,6 +24,10 @@ jobs:
   analyze:
     name: Analyze
     runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
```

Comment:
nit: `actions` and `contents` are only required for private repository. 

See the `Usage` section in https://github.com/github/codeql-action

Reply:
I think permissions should reflect whole set of permissions required, even those that are given by default in certain environments.